src/controllers/category.controller.ts

```diff
@@ -0,0 +1,303 @@
+    ],
+  })
+  async postCategory(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodCategory_Input',
+          },
+          examples: {
+            'Create a Category': {
+              value: {
+                name: 'Mental Health',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    })
+    category: Omit<GoodCategory, 'id'>,
+  ): Promise<unknown> {
+    return this.repository.create(category);
+  }
+
+  /**
+   * Update a Category's details
+   *
+   * @param id The PoG ID of the category
+   * @param category
+   */
+  @put('/category/{id}', {
+    summary: 'Change Category Details',
+    operationId: 'put-category',
+    responses: {
+      '200': {
+        description: 'Category Updated',
+      },
+      '401': {
+        description: 'Unauthorized',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+        headers: {},
+      },
+      '404': {
+        description: 'Category Not Found',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+    },
+    requestBody: {
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodCategory_Input',
+          },
+          examples: {
+            'Change Category Details': {
+              value: {
+                name: 'Animal Welfare',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    },
+    description: "Update a Category's details",
+    security: [
+      {
+        Category_API_Key: [],
+      },
+    ],
+    parameters: [
+      {
+        schema: {
+          type: 'integer',
+          example: 3,
+        },
+        name: 'id',
+        in: 'path',
+        required: true,
+        description: 'The PoG ID of the category',
+      },
+    ],
+  })
+  async putCategory(
+    @param({
+      schema: {
+        type: 'integer',
+        example: 3,
+      },
+      name: 'id',
+      in: 'path',
+      required: true,
+      description: 'The PoG ID of the category',
+    })
+    id: number,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodCategory_Input',
+          },
+          examples: {
+            'Change Category Details': {
+              value: {
+                name: 'Animal Welfare',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    })
+    category: GoodCategoryInput | any,
+  ): Promise<unknown> {
+    delete category?.id;
+    return this.repository.updateById(id, category).catch((err: Error) => {
+      if (err.message == '404')
```

Comment:
```suggestion
      if (err.message == 'Document not found')
```
to stay consistent with the error handling of BetterFirestoreConnector